File: examples/Python/tornado_ioloop/hwserver.py
#!/usr/bin/env python

"""
synopsis:
    Hello World server in Python.
    Binds REP socket to Url.
    Expects b"Hello" from client, replies with b"World".
    Modified for tornado/ioloop: Dave Kuhlman <dkuhlman(at)davekuhlman(dot)org>
usage:
    python hwserver.py
"""

import sys
import zmq
from zmq.eventloop.future import Context
from zmq.eventloop.ioloop import IOLoop
from tornado import gen


Url = 'tcp://127.0.0.1:5555'
Ctx = Context()


@gen.coroutine
def run():
    print("Getting ready for hello world client.  Ctrl-C to exit.\n")
    socket = Ctx.socket(zmq.REP)
    socket.bind(Url)
    while True:
        #  Wait for next request from client
        message = yield socket.recv()
        print("Received request: {}".format(message))
        #  Do some 'work'
        yield gen.sleep(1)
        #  Send reply back to client
        message = message.decode('utf-8')
        message = '{}, world'.format(message)
        message = message.encode('utf-8')
        print("Sending reply: {}".format(message))
        yield socket.send(message)


def main():
    args = sys.argv[1:]
    if len(args) != 0:
        sys.exit(__doc__)
    try:
        loop = IOLoop.current()
        loop.add_callback(lambda: run())
        loop.start()
    except KeyboardInterrupt:
        print('\nFinished (interrupted)')
        sys.exit(0)


if __name__ == '__main__':
    main()

File: examples/Java/src/main/java/com/imatix/zguide/ex1_1/HwServer.java
package com.imatix.zguide.ex1_1;

import org.zeromq.ZMQ.Context;
import org.zeromq.ZMQ.Socket;

import static com.imatix.zguide.commons.ZConstants.URL.TCP;
import static java.lang.String.format;
import static org.zeromq.ZMQ.REP;
import static org.zeromq.ZMQ.context;

/**
 * Hello World server in Java
 * Binds REP socket to tcp://*:5555
 * Expects "Hello" from client, replies with "World"
 *
 * @since 1.0
 */

public class HwServer {

    public static void main(String[] args) throws InterruptedException {
        Context context = context(1);
        // Socket to Talk to Clients
        Socket responder = context.socket(REP);
        responder.bind(TCP);
        while (!Thread.currentThread().isInterrupted()) {
            // Wait for next request from clients
            byte[] request = responder.recv(0);
            System.out.println(format("Received  %s", new String(request)));
            //Do Some business
            Thread.sleep(1000);
            // Send reply back to client
            responder.send("World".getBytes(), 0);
        }
        responder.close();
        context.term();
    }
}
